x/gamm/keeper/migrate.go

```diff
@@ -140,6 +159,49 @@ func (k Keeper) OverwriteMigrationRecords(ctx sdk.Context, migrationInfo types.M
 	}
 }
 
+// redirectDistributionRecord redirects the distribution record for the given balancer pool to the given concentrated pool.
+func (k Keeper) redirectDistributionRecord(ctx sdk.Context, cfmmPoolId, clPoolId uint64) error {
+	// Get CFMM gauges
+	cfmmGauges, err := k.poolIncentivesKeeper.GetGaugesForCFMMPool(ctx, cfmmPoolId)
+	if err != nil {
+		return err
+	}
+
+	if len(cfmmGauges) == 0 {
+		return fmt.Errorf("no gauges found for cfmm pool %d", cfmmPoolId)
+	}
+
+	// Get longest gauge duration from CFMM pool.
+	longestDurationGauge := cfmmGauges[0]
+	for i := 1; i < len(cfmmGauges); i++ {
+		if cfmmGauges[i].DistributeTo.Duration > longestDurationGauge.DistributeTo.Duration {
+			longestDurationGauge = cfmmGauges[i]
+		}
+	}
+
+	// Get concentrated liquidity gauge duration.
+	distributionEpochDuration := k.incentivesKeeper.GetEpochInfo(ctx).Duration
+
+	// Get concentrated gauge corresponding to the distribution epoch duration.
+	concentratedGaugeId, err := k.poolIncentivesKeeper.GetPoolGaugeId(ctx, clPoolId, distributionEpochDuration)
+	if err != nil {
+		return err
+	}
+
+	// Iterate through all the distr records, and redirect the old balancer gauge to the new concentrated gauge.
+	distrInfo := k.poolIncentivesKeeper.GetDistrInfo(ctx)
+	for i, distrRecord := range distrInfo.Records {
+		if distrRecord.GaugeId == longestDurationGauge.Id {
+			distrInfo.Records[i].GaugeId = concentratedGaugeId
```

Comment:
Let's make sure to let Johnny know that we're mutating existing records so future gov props don't accidentally create duplicate incentives

Reply:
Let him know, thanks!